src/Strategy/Ruby/Parse.hs

```diff
@@ -29,6 +36,22 @@ selectDelim = \case
   '[' -> ('[', ']')
   c -> (c, c)
 
+betweenDelim :: (Char, Char) -> Parser Text
+betweenDelim (d1, d2) =
+  mconcat
+    <$> between
+      (char d1)
+      (char d2)
+      ( many
+          ( try delimEscape
+              <|> (toText <$> anySingleBut d2)
+          )
+      )
+  where
+    -- only the close delimiter can stop parsing, so only check
+    -- for escaped versions of that.
+    delimEscape = string $ "\\" <> toText d2
```

Comment:
I think the intent here is to consume an escaped `d2` before `between` can reach it (and terminate the parse), correct?

Reply:
That is correct. Looking at this after a few days with fresh eyes I'm wondering if there's a simpler way to do it, do you have any ideas?

Reply:
I don't; this is how I'd do it too.
I think it would be useful to comment that this is the intent though to make it more clear to future readers.